### PR TITLE
Improve SdkTestContext

### PR DESCRIPTION
**Pull request by rtso**
### TL;DR
Enhanced the SdkTestContext to simplify test setup and improve transaction handling in the indexer processor testing framework.

### What changed?
- Refactored `SdkTestContext` to store and manage request versions internally
- Make `SdkTestContext` fields private so devs don't accidentally use `transaction_batches` incorrectly
- Added new helper methods `get_request_start_version()` and `get_transactions_count()` that will be used in processor tests
- Split server initialization into a separate `init_mock_grpc()` method. This is to make the constructor non-async which is generally encouraged. 
- Simplified `create_transaction_stream_config()` by removing manual version parameters

### Testing
cargo test

### CODE REVIEW

**Comment by rtso** [comment]
* **#82** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-indexer-processor-sdk/82?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rtso and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by yuunlimm — APPROVED** [review]
